tests/choice/run.py

```diff
@@ -1,92 +1,82 @@
-  for i in ["case_one_container", "case_two_container"]:
-    assert hasattr(t.container, i), "object does not have choice container" + \
-        ", %s" % i
-  for i in ["choice_one", "choice_two", "case_one", "case_two"]:
-    assert not hasattr(t.container, i), \
-        "object has an erroneous choice option, %s" % i
-
-  assert t.container.case_one_container.case_one_leaf == 0, \
-      "object does not have the correct value for case_one_leaf, %s" % \
-          t.container.case_one_container.case_one_leaf
-  assert t.container.case_two_container.case_two_leaf == 0, \
-      "object does not have the correct value for case_two_leaf, %s" % \
-          t.container.case_two_container.case_two_leaf
-
-  t.container.case_one_container.case_one_leaf = 42
-  assert t.container.case_one_container.case_one_leaf == 42, \
-      "object did not specify a value within the choice correctly, %s" \
-          % t.container.case_one_container.case_one_leaf
-  assert t.container.case_two_container.case_two_leaf == 0, \
-      "object erroneously set another value wtihin the choice, %s" \
-        % t.container.case_two_container.case_two_leaf
-
-  t.container.case_two_container.case_two_leaf = 42
-  assert t.container.case_two_container.case_two_leaf == 42, \
-      "object did not allow the other half of the choice to be specified, %s" \
-        % t.container.case_two_container.case_two_leaf
-  assert t.container.case_one_container.case_one_leaf == 0, \
-    "object did not reset the value of the case-one side of the choice, %s" \
-       % t.container.case_one_container.case_one_leaf
-
-  t.container.case_one_container.user.add("first")
-  assert t.container.case_one_container.user["first"].username == "first", \
-       "object has the wrong username for user in case one list"
-
-  t.container.case_two_container.user.add("second")
-  assert t.container.case_two_container.user["second"].username == "second", \
-       "object has the wrong username for user in case two list"
-  assert len(t.container.case_one_container.user.keys()) == 0, \
-       "Adding to the second user list did not remove entries from the first"
-
-  if not keepfiles:
-    os.system("/bin/rm %s/bindings.py" % this_dir)
-    os.system("/bin/rm %s/bindings.pyc" % this_dir)
+import unittest
+
+from tests.base import PyangBindTestCase
+
+
+class ChoicesTests(PyangBindTestCase):
+  yang_files = ['choice.yang']
+
+  def setUp(self):
+    self.choice_obj = self.bindings.choice()
+
+  def test_class_has_container(self):
+    self.assertTrue(hasattr(self.choice_obj, 'container'), "Object does not have container")
+
+  def test_class_has_choice_containers(self):
+    for container in ["case_one_container", "case_two_container"]:
+      with self.subTest(container=container):
+        self.assertTrue(hasattr(self.choice_obj.container, container),
+          "Object does not have choice container %s" % container)
+
+  def test_class_does_not_have_unknown_choices(self):
```

Comment:
Suggest to rename to `test_class_does_not_have_choices_as_attributes` - the intention here is to check that we don't generate `choice-one` which doesn't include in the YANG schema path.

i.e.,

```
container ca {
  choice a-choice {
     case a { 
       leaf ch-one { type string; }
     }
     case b { 
       leaf ch-two { type string; }
    }
  }
}
```

results in schema paths that are `/ca/ch-one` and `/ca/ch-two`, so we want to make sure that the `a-choice`, `a` and `b` statements don't get erroneously translated into an attribute of `ca`.

Reply:
Could you address this one before we merge? I think the existing name is not quite clear.